src/main/java/org/opentripplanner/transit/model/network/TripPattern.java

```diff
@@ -56,29 +56,56 @@ public final class TripPattern
   private static final Logger LOG = LoggerFactory.getLogger(TripPattern.class);
 
   private final Route route;
+
   /**
-   * The stop-pattern help us reuse the same stops in several trip-patterns; Hence saving memory.
-   * The field should not be accessible outside the class, and all access is done through method
-   * delegation, like the {@link #numberOfStops()} and {@link #canBoard(int)} methods.
+   * The Route and StopPattern together form the primary key of the TripPattern. They are the shared
+   * set of characteristics that group many trips together into one TripPattern. This grouping saves
+   * memory by not replicating any details shared across all trips in the TripPattern, but it is
+   * also essential to some optimizations in routing algorithms like Raptor.
+   * <p>
+   * This field should not be accessed outside this class. All access to the StopPattern is
+   * performed through method  delegation, like the {@link #numberOfStops()} and
+   * {@link #canBoard(int)} methods.
    */
   private final StopPattern stopPattern;
+
+  /**
+   * TripPatterns hold a reference to a Timetable (i.e. TripTimes for all Trips in the pattern) for
+   * only scheduled trips from the GTFS or NeTEx data. If any trips were later updated in real time,
+   * there will be another Timetable holding those updates and reading through to the scheduled one.
+   * This realtime Timetable is retrieved from a TimetableSnapshot.
+   * Also see end of Javadoc on TimetableSnapshot for more details.
```

Comment:
I fully agree about where documentation should be in the final state. But my intent in this PR was to include temporary documentation that would group information together to help guide refactoring (and clarify the current approach). The comments here are highlighting the fact that this field of TripPattern contains only scheduled timetables and not realtime. It's (non-obviously) hinting that this should be changed, and how it can be changed. I'd say this documentation is a bit long and awkward because the system it's describing is more complicated and awkward than it should be, and the documentation will become simpler when the system is simplified.

For now I will keep this information together on this field, but will rephrase it to make it clear the intent is to change the approach, not freeze the current system and document it in place.